src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -204,10 +211,14 @@ var createFiber = function(tag : TypeOfWork, key : null | string) : Fiber {
     alternate: null,
 
   };
+
   if (__DEV__) {
     (fiber : any)._debugID = debugCounter++;
+    (fiber : any)._debugSource = null;
+    (fiber : any)._debugOwner = null;
   }
-  return fiber;
+
+  return (fiber : any); // Don't set dev-only fields in production
```

Comment:
I actually didn't need it anymore because I annotated fields themselves as optional. Fixed.